theano/tensor/nnet/ctc.py

```diff
@@ -0,0 +1,240 @@
+from __future__ import (division, absolute_import, print_function)
+import os
+import os.path
+import theano.tensor as T
+from theano import config
+from theano import gof
+from theano.gof import local_optimizer
+from theano.gof.cmodule import GCC_compiler
+from theano.tensor.opt import register_canonicalize
+from theano.tensor.opt import register_stabilize
+from theano.tensor.extra_ops import cpu_contiguous
+from theano.gradient import grad_undefined
+
+
+def _ctc_find_lib():
+    """
+    Find the directory that contains libwarpctc.so
+    """
+    lib_found = False
+    for lib_dir in ["build", "lib", "lib64"]:
+        lib_path = os.path.join(config.ctc.root, lib_dir)
+        if os.path.isdir(lib_path) and os.path.exists(lib_path):
+            lib_found = os.path.exists(os.path.join(lib_path, "libwarpctc.so"))
+            if lib_found:
+                return True, lib_path
+    return False, None
```

Comment:
Hello, @joaovictortr ! While installing CTC on jenkins, we remarked that user will necessarly have to specify `ctc.root` even if CTC is installed in default system paths (e.g. in `/usr/local` on Linux). We should be able to automatically find CTC if user installed it as a default system library.

I suggest that:
1) `_ctc_find_lib()` just returns either `lib_path` if it is found, or `None` otherwise.
2) `ctc_present()` passes the result of `_ctc_find_lib()` directly to `_ctc_check_compile()` without any checking on `ctc_lib_path`.
3) Then, in `_ctc_check_compile()`, you add `-L<libpath>` only if libpathh is not None. Otherwise, GCC will try to find `libwarpctc.so` in default systems paths.